components/core/src/os/net/mod.rs

```diff
@@ -22,3 +22,71 @@ mod imp;
 mod imp;
 
 pub use self::imp::*;
+
+extern crate dns_lookup;
+
+use dns_lookup::{getaddrinfo,
+                 AddrInfoHints};
+use std::io;
+
+// lookup_fqdn returns a vector of fqdn that resolves the provided hostname.
+//
+// Since the underlying crate is platform agnostic, this function is as well,
+// we only implement a single function `ai_canonname()` that returns the right
+// hint flag for the running operating system.
+pub fn lookup_fqdn(hostname: &str) -> io::Result<Vec<String>> {
+    let hints = AddrInfoHints { flags: ai_canonname(),
+                                ..AddrInfoHints::default() };
+    let addrinfos =
+        getaddrinfo(Some(hostname), None, Some(hints))?.collect::<std::io::Result<Vec<_>>>()?;
+
+    let canonnames = addrinfos.into_iter()
+                              .filter_map(|info| info.canonname)
+                              .collect();
+
+    Ok(canonnames)
+}
+
+// fqdn returns the fully qualified domain name of the running machine
+pub fn fqdn() -> Option<String> {
+    // Implementation 1 - Using match statements
+    match hostname() {
+        Ok(hostname) => {
+            match lookup_fqdn(&hostname) {
+                Ok(fqdns) => fqdns.into_iter().find(|ref h| h.contains(&hostname)),
+                // @afiune if the lookup_fqdn returns an Err(), should we
+                // return the hostname instead of None?
+                Err(_) => None,
+            }
+        }
+        Err(_) => None,
+    }
+}
+
+#[test]
+#[ignore]
+fn test_fqdn() {
+    // @afiune This test is ignore because it is testing the actual
+    // fqdn of the running machine, mine has 'afiune-ubuntu-vb.lala.com'
```

Comment:
I think @christophermaier was trying to use the `resolv_wrapper` tool so we can mock the hostname of the machine. I just left this test since I used it to test the implementation and as a reminder that we actually need to test. If you think is worthless I don't mind deleting it.

Reply:
Yeah, let's remove

Reply:
Ok.